scripts/install/opt/seagate/cortx/motr/bin/motr_setup

```diff
@@ -0,0 +1,192 @@
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+        args = parser.parse_args(argv)
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """Add Command args for parsing."""
+
+        parser1 = parser.add_parser(cls.name, help='setup %s' % name)
+        parser1.add_argument('url', help='Config URL')
+        parser1.add_argument('args', nargs='*', default=[], help='args')
+        parser1.set_defaults(command=cls)
+
+
+class PostInstallCmd(Cmd):
+    """PostInstall Setup Cmd"""
+    name = "post_install"
+
+    def __init__(self, args: dict):
+        super().__init__(args)
+
+    def process(self):
+        print("Processing %s %s %s" %(self.name, self.url, self._args))
+        validate_motr_rpm(self)
+
+class ConfigCmd(Cmd):
+    """Config Setup Cmd"""
+    name = "config"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        print("Processing %s %s %s" %(self.name, self.url, self._args))
+        motr_config(self)
+
+class InitCmd(Cmd):
+    """Init Setup Cmd"""
+    name = "init"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        print("Processing %s %s %s" %(self.name, self.url, self._args))
+        config_lvm(self)
+        configure_net(self)
+
+class TestCmd(Cmd):
+    """Test Setup Cmd"""
+    name = "test"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        print("Processing %s %s %s" %(self.name, self.url, self._args))
+        test_lnet(self)
+
+class ResetCmd(Cmd):
+    """Reset Setup Cmd"""
+    name = "reset"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        pass
+
+class CleanupCmd(Cmd):
+    """Cleanup Setup Cmd"""
+    name = "cleanup"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        pass
+
+class UpgradeCmd(Cmd):
+    """Upgrade Setup Cmd"""
+    name = "upgrade"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        pass
+
+class BackupCmd(Cmd):
+    """Backup Setup Cmd"""
+    name = "backup"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        pass
+
+class RestoreCmd(Cmd):
+    """Restore Setup Cmd"""
+    name = "restore"
+
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        pass
+
+def main(argv: dict):
+    try:
+        desc = "Motr Setup command"
+        command = Cmd.get_command(desc, argv[1:])
+        command.process()
+
+    except Exception as e:
+        sys.stderr.write("error: %s\n\n" % str(e))
+        sys.stderr.write("%s\n" % traceback.format_exc())
+        Cmd.usage(argv[0])
+        return errno.EINVAL
+
+if __name__ == '__main__':
```

Comment:
A series cmd processing classes are defined: setup, config, restore, ...
But can you please give some example about how to use them?
What is the correct sequence to call these commands and what parameters should be provided.